### PR TITLE
DT - 23599 Applying FeatureFlag for testing Find Forms

### PR DESCRIPTION
**Pull request by zacharymorel**
## Issue
https://github.com/department-of-veterans-affairs/va.gov-team/issues/23599

## Description
Adding code test out the new Lighthouse search changes in staging.va.gov.
Previous PR Related => https://github.com/department-of-veterans-affairs/vets-website/pull/17039

## Testing done
Adjusted Unit tests, ran e2e test, tested locally via switching the flags on and off.

## Screenshots
Screen Shot 2021-05-05 at 9 58 14 AM

## Acceptance criteria
- [x] Add code to consume the Feature Flag so testing can happen on staging

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/find-forms/constants/index.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const FETCH_FORMS = 'findVAForms/FETCH_FORMS';
 export const FETCH_FORMS_FAILURE = 'findVAForms/FETCH_FORMS_FAILURE';
 export const FETCH_FORMS_SUCCESS = 'findVAForms/FETCH_FORMS_SUCCESS';
 export const UPDATE_PAGINATION = 'findVAForms/UPDATE_PAGINATION';
+export const TEST_OPTION_RELEVANCE = 'Relevance';
```
Comment (kelsonic) [inline src/applications/find-forms/constants/index.js:7]: Can we scope this to `findVAForms`?

File: src/applications/find-forms/containers/SearchResults.jsx [hunk 2]
```diff
@@ -39,10 +43,17 @@ export class SearchResults extends Component {
...
     if (justRefreshed) {
       focusElement('[data-forms-focus]');
     }
+
+    // NOTE: This is only for testing Lighthouse Search Algorithm
+    if (
+      previousProps.useLighthouseSearchAlgo === undefined &&
+      props.useLighthouseSearchAlgo === true
+    ) {
+      props.updateSortByPropertyName(TEST_OPTION_RELEVANCE, this.props.results);
```
Comment (kelsonic) [inline src/applications/find-forms/containers/SearchResults.jsx:55]:
```suggestion
      props.updateSortByPropertyName(TEST_OPTION_RELEVANCE, props.results);
```

File: src/applications/find-forms/containers/SearchForm.jsx [hunk 3]
```diff
@@ -52,8 +52,8 @@ export class SearchForm extends Component {
 
   onSubmitHandler = event => {
     event.preventDefault();
-    const { useSearchQueryAutoCorrect } = this.props;
-    this.props.fetchFormsThunk(this.state.query, { useSearchQueryAutoCorrect });
+    const { useLighthouseSearchAlgo } = this.props;
+    this.props.fetchFormsThunk(this.state.query, { useLighthouseSearchAlgo });
```
Comment (kelsonic) [inline src/applications/find-forms/containers/SearchForm.jsx:56]:
```suggestion
    const { fetchFormsThunk, useLighthouseSearchAlgo } = this.props;
    fetchFormsThunk(this.state.query, { useLighthouseSearchAlgo });
```

File: src/applications/find-forms/containers/SearchResults.jsx [hunk 4]
```diff
@@ -149,6 +161,11 @@ export class SearchResults extends Component {
       );
     }
 
+    // Derive sort options
+    const sortOptions = useLighthouseSearchAlgo
+      ? [TEST_OPTION_RELEVANCE, ...SORT_OPTIONS]
+      : SORT_OPTIONS;
```
Comment (kelsonic) [inline src/applications/find-forms/containers/SearchResults.jsx:167]: Maybe call this constant `DEFAULT_SORT_OPTIONS` instead of `SORT_OPTIONS`?

**Review by kelsonic — APPROVED** [review]
LGTM 🙂

**Review by U-DON — APPROVED** [review]

File: src/applications/find-forms/containers/SearchResults.jsx (hunk 4 above)
Comment (zacharymorel) [reply src/applications/find-forms/containers/SearchResults.jsx:167]: Ah great point sir!

**Pull request merged**